src/arch/xen/boot.S

```diff
@@ -57,21 +55,261 @@ C_ENTRY(_start):
 	addl    $0xc, %esp
 1:
 
-	//call    gdt_init
-	//call    idt_init
-
+#if 0 // TODO smp is not supported on Xen
 #if NCPU > 1
 	call    cpudata_init
 #endif /* NCPU > 1 */
+#endif
 
-/* we already save eax and ebx on the stack */
-	//call    multiboot_save_info
-	addl    $8, %esp
+	push 	%esi
+	call    xen_kernel_start /* call C code */
 
+	.org 0x1000
+	.p2align 12
+C_LABEL(xen_shared_info):
+	.org 0x2000
 
```

Comment:
p2align line seems redundant, others are required, these labels and skip 2 pages, at which Xen will place memory-mapped interface to hypervisor.
Some cleanup will not harm, will perfrom